src/forms/Rfc003ParamsForm.tsx

```diff
@@ -59,80 +60,72 @@ function Rfc003ParamsForm({
     return expiry !== "" ? parseInt(expiry, 10) : undefined;
   };
 
-  return (
-    <React.Fragment>
+  function LedgerFields(
+    ledger: string,
+    expiryKey: string,
+    identityKey: string,
+    expiry: number,
+    identity: string | undefined
+  ) {
+    const expiryLabel = titleCase(expiryKey);
+    const identityLabel = titleCase(identityKey);
+
+    return (
       <Grid item={true} xs={12} container={true} spacing={0}>
         <Grid item={true} md={6} xs={12}>
           <TextField
```

Comment:
:wrench: I think a component that only parameterizes the `<TextField>` would be more useful. E.g. Remove the `Grid` components from this one and compose them outside again.

Then you can get rid of the computation of the `data-cy` attribute since you can just set it on the very outside where you know, which fields you want to create.